src/Microsoft.Dnx.Runtime/Internal/ConcurrencyUtilities.cs

```diff
@@ -10,6 +11,11 @@ namespace Microsoft.Dnx.Runtime.Internal
 {
     public static class ConcurrencyUtilities
     {
+#if DNXCORE50
+        private static ConcurrentDictionary<string, System.Threading.Semaphore> _nameSemaphore =
```

Comment:
We tested restoring for Mvc repo and the peak memory usage was about 500MB. Yes, we should remove entries. Working on it.